lib/entities/organization-membership.ts

```diff
@@ -0,0 +1,127 @@
+import cloneDeep from 'lodash/cloneDeep'
+import { freezeSys, toPlainObject } from 'contentful-sdk-core'
+import enhanceWithMethods from '../enhance-with-methods'
+import errorHandler from '../error-handler'
+import { createDeleteEntity } from '../instance-actions'
+import { AxiosInstance } from 'axios'
+import { MetaSysProps, QueryOptions, DefaultElements, CollectionProp } from '../common-types'
+
+export interface Options {
+  query?: QueryOptions
+}
+export type OrganizationMembershipProps = {
+  /**
+   * System metadata
+   */
+  sys: MetaSysProps
+
+  /**
+   * Role
+   */
+  role: string
+
+  /**
+   * status
+   */
+  status: boolean
+}
+
+export interface OrganizationMembership
+  extends OrganizationMembershipProps,
+    DefaultElements<OrganizationMembershipProps> {
+  /**
+   * Sends an update to the server with any changes made to the object's properties
+   * @memberof OrganizationMembership
+   * @func update
+   * @return {Promise<OrganizationMembership>} Object returned from the server with updated changes.
+   * @example
+   * const contentful = require('contentful-management')
+   *
+   * const client = contentful.createClient({
+   *   accessToken: '<content_management_api_key>'
+   * })
+   *
+   * client.getOrganization('organization_id')
+   * .then(org => org.getOrganizationMembership('organizationMembership_id'))
+   * .then((organizationMembership) => {
+   *  organizationMembership.role = 'member';
+   *  organizationMembership.update();
+   * })
+   * .catch(console.error)
+   */
+  update(): Promise<OrganizationMembership>
+
+  /**
+   * Deletes this object on the server.
+   * @example```javascript
+   * const contentful = require('contentful-management')
+   *
+   * const client = contentful.createClient({
+   *   accessToken: '<content_management_api_key>'
+   * })
+   *
+   * client.getOrganization('organization_id')
+   * .then(org => org.getOrganizationMembership('organizationMembership_id'))
+   * .then((organizationMembership) => {
+   *  organizationMembership.delete();
+   * })
+   * .catch(console.error)
+   * ```
+   */
+  delete(): Promise<void>
+}
+
+function createOrganizationMembershipApi(http: AxiosInstance) {
+  return {
+    update: function () {
+      const raw = this.toPlainObject()
```

Comment:
In order to make `this` typed, I suggest using the following approach: https://github.com/contentful/contentful-management.js/blob/master/lib/entities/app-installation.ts#L67-L73

Reply:
@suevalov done! :)